tests/EStimLibrary.UnitTests/Core/HardwareInterfaces/LeadTests.cs

```diff
@@ -0,0 +1,618 @@
+using EStimLibrary.Core;
+using EStimLibrary.Core.HardwareInterfaces;
+using Xunit;
+using System.Collections.Generic;
+
+namespace EStimLibrary.UnitTests.Core.HardwareInterfaces
+{
+    public class LeadTests
+    {
+        /// <summary>
+        /// Test the parameterized constructor with different data values.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(ConstructorTestData))]
+        public void ConstructorShouldInitSortedSetsAndDirection(SortedSet<int> contactSet,
+            SortedSet<int> outputSet, Constants.CurrentDirection currentDirection)
+        {
+            // Act
+            var lead = new Lead(contactSet, outputSet, currentDirection);
+
+            // Assert
+            Assert.Equal(contactSet, lead.ContactSet);
+            Assert.Equal(outputSet, lead.OutputSet);
+            Assert.Equal(currentDirection, lead.CurrentDirection);
+        }
+
+        public static IEnumerable<object[]> ConstructorTestData()
+        {
+            return new List<object[]>
+            {
+                new object[]
+                {
+                    new SortedSet<int> { 1, 2, 3 },
+                    new SortedSet<int> { 0, 2, 3 },
+                    Constants.CurrentDirection.SOURCE
+                },
+                new object[]
+                {
+                    new SortedSet<int> { 4, 5, 6 },
+                    new SortedSet<int> { 1, 2 },
+                    Constants.CurrentDirection.SOURCE
+                },
+                new object[]
+                {
+                    new SortedSet<int> { 7, 8, 9 },
+                    new SortedSet<int> { 3, 5, 7 },
+                    Constants.CurrentDirection.SINK
+                }
+            };
+        }
+
+        // Test for GetConnectedOutputs
+        [Theory]
+        [MemberData(nameof(GetConnectedOutputsTestData))]
+        public void GetConnectedOutputs_ShouldReturnExpectedResults(int id, bool searchIsAnOutput,
+            Lead lead, SortedSet<int> expectedOutputs, bool expectedResult)
+        {
+            // Arrange
+            // Act
+            var result = lead.GetConnectedOutputs(id, searchIsAnOutput, out var connectedOutputs);
+
+            // Assert
+            Assert.Equal(expectedResult, result);
+            Assert.Equal(expectedOutputs, connectedOutputs);
+        }
+
+        public static IEnumerable<object[]> GetConnectedOutputsTestData()
+        {
+            return new List<object[]>
+            {
+                // Testing with an output as input when that output is connected to the lead
+                // We expect true and a modified output set
```

Comment:
Tiny note: instead of "a modified" would just comment "a remaining" or "a non-inclusive" or something like that so doesn't misconstrue that the method is modifying internal data when it shouldn't be